packages/server/src/serverConnection.ts

```diff
@@ -116,6 +116,9 @@ export default class ServerConnection implements IServerConnection {
     token: CancellationToken
   ): Promise<Result<Void, FxError>> {
     const corrId = inputs.correlationId ? inputs.correlationId : "";
+    if (inputs.env && inputs.env === environmentManager.getLocalEnvName()) {
```

Comment:
Could this change be in VS extension - set `inputs.workflowFilePath` before calling `server.provisionResourcesRequest`.